repos/system_upgrade/el7toel8/actors/checkpcidrivers/actor.py

```diff
@@ -0,0 +1,155 @@
+        driver
+        for driver, data in restricted_driver_names_on_host.items()
+        if data["available_rhel8"] == 0
+    )
+    unavailable_pci_ids = ",\n".join(
+        pci_id
+        for pci_id, data in restricted_pci_ids_on_host.items()
+        if data["available_rhel8"] == 0
+    )
+    unsupported_driver_names = ",\n".join(
+        driver
+        for driver, data in restricted_driver_names_on_host.items()
+        if data["supported_rhel8"] == 0
+    )
+    unsupported_pci_ids = ",\n".join(
+        pci_id
+        for pci_id, data in restricted_pci_ids_on_host.items()
+        if data["supported_rhel8"] == 0
+    )
+    title = "Detected PCI drivers that are restricted on RHEL8."
+    summary = (
+        "There are following drivers that are restricted on RHEL 8 system.\n\n"
+        "Host devices which are unavailable on RHEL8:\n {}{}\n\n"
+        "Host devices which are unsupported on RHEL8:\n {}{}\n\n"
+    ).format(
+        unavailable_driver_names,
+        unavailable_pci_ids,
+        unsupported_driver_names,
+        unsupported_pci_ids,
+    )
+    reports = [
+        reporting.Title(title),
+        reporting.Summary(summary),
+        reporting.Severity(reporting.Severity.HIGH),
+        reporting.Tags([reporting.Tags.KERNEL, reporting.Tags.DRIVERS]),
+    ]
+    if inhibit_upgrade:
+        reports += [reporting.Flags([reporting.Flags.INHIBITOR])]
+    return reports
+
+
+class CheckPCIDrivers(Actor):
+    """Check if host PCI devices drivers has restrictions on target system.
+
+    If the driver is unavailable on a target system - then the upgrade will
+        be inhibited
+    If the driver is available, but not supported - then the upgrade will
+        continue, but the relevant report will be generated
+    If the driver is available and supported - then the upgrade will continue
+        normally
+    """
+
+    name = "check_pci_drivers"
+    consumes = (
+        PCIDevices,
+        RestrictedPCIDevices,
+    )
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        pci_devs = next(self.consume(PCIDevices))
+        try:
+            restricted_pci_devs = next(self.consume(RestrictedPCIDevices))
+        except StopIteration:
+            # when there is no restricted pci devices - it is fine and we
+            #   just do nothing
+            self.log.info(
+                "No data about restricted PCI devices. Skipping" " the actor."
+            )
+            return
+        else:
+            driver_names = tuple(dev.driver for dev in pci_devs.devices)
+            pci_ids = tuple(dev.pci_id for dev in pci_devs.devices)
+
+            restricted_driver_names = restricted_pci_devs.driver_names[
+                "devices"
+            ]
+            restricted_pci_ids = restricted_pci_devs.pci_ids["devices"]
+
+            restricted_driver_names_on_host = dict(
+                filter(  # pylint:disable=W0110
```

Comment:
The linter fails on py2 with:
```
[W0110 deprecated-lambda] map/filter on lambda could be replaced by comprehension File: repos/system_upgrade/el7toel8/actors/checkpcidrivers/actor.py, line 109, in CheckPCIDrivers.process

[W0110 deprecated-lambda] map/filter on lambda could be replaced by comprehension File: repos/system_upgrade/el7toel8/actors/checkpcidrivers/actor.py, line 116, in CheckPCIDrivers.process
```
basically, I like this construction more. It has some performance benefits on  py3 and works on py2. A kind of a matter of taste in case of readability.

I can't find something relevant in our guidelines